### PR TITLE
Clear next method when clearing TIs

### PR DESCRIPTION
**Pull request by tanelk**
Clearing TIs does not clear deferred TIs `next_method` field.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by uranusjr** [comment]
Same as #19640 (which unfortunately went stale)

**Review by uranusjr — APPROVED** [review]

File: tests/models/test_cleartasks.py [hunk 1]
```diff
@@ -106,6 +106,29 @@ def test_clear_task_instances_external_executor_id(self, dag_maker):
...
+        ) as dag:
+            EmptyOperator(task_id='task0')
+
+        ti0 = dag_maker.create_dagrun().task_instances[0]
+        ti0.state = State.DEFERRED
+        ti0.next_method = "next_method"
+        ti0.next_kwargs = {}
+
+        session.add(ti0)
+        session.commit()
```
Comment (uranusjr) [inline tests/models/test_cleartasks.py:123]: Would `flush` be enough here?

**Comment by tanelk** [comment]
Pinging @ashb for a review of this bugfix.

**Comment by tanelk** [comment]
@uranusjr could this approved PR get merged?

**Review by potiuk — APPROVED** [review]

**Pull request merged**